tests/dip_sip/test_dip_sip.py

```diff
@@ -0,0 +1,168 @@
+import pytest
+import ptf.testutils as testutils
+from ipaddress import ip_address
+import logging
+
+TOPO_LIST = {'t0', 't1', 't1-lag'}
+PORTS_TOPO = {'t1'}
+LAG_TOPO = {'t0', 't1-lag'}
+DEFAULT_HLIM_TTL = 64
+WAIT_EXPECTED_PACKET_TIMEOUT = 5
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='function', autouse=True)
+def prepare_ptf(testbed_devices):
+    ptfhost = testbed_devices["ptf"]
+    # remove existing IPs from ptf host
+    ptfhost.script('scripts/remove_ip.sh')
+    # set unique MACs to ptf interfaces
+    ptfhost.script('scripts/change_mac.sh')
+
+
+def lag_facts(dut):
+    facts = {}
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
+    if not mg_facts['minigraph_portchannels']:
+        pytest.fail("minigraph_portchannels is not defined")
+    host_facts = dut.setup()['ansible_facts']
+    # minigraph facts
+    src_lag = mg_facts['minigraph_portchannel_interfaces'][2]['attachto']
+    dst_lag = mg_facts['minigraph_portchannel_interfaces'][0]['attachto']
+    facts['src_port'] = src_lag
+    facts['dst_port'] = dst_lag
+    logger.info("src_lag is {}, dst_lag is {}".format(src_lag, dst_lag))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][src_lag]['members'][0]]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_lag]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_lag]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_lag]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_lag]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][src_lag]['members'][0]]]
+
+    return facts
+
+
+def port_facts(dut):
```

Comment:
suggest to move to common? @wangxin , @yxieca

Reply:
According to implementation of this function, it looks more test specific than common behavior. I think it is OK to keep it here.